src/pages/Armies/ViewArmyPage.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PageTitle from '../../atoms/PageTitle';
+import store from '../../Redux/store';
+import PageLoading from '../../atoms/PageLoading';
+import TextRow from '../../atoms/TextRow';
+import findRealSquadName from '../../utilities/findRealSquadName';
+import findRealUnitName from '../../utilities/findRealUnitName';
+import findRealEquipmentName from '../../utilities/findRealEquipmentName';
+import findUnitAbilities from '../../utilities/findUnitAbilities';
+import calcSquadNumber from '../../utilities/calcSquadNumber';
+import RedirectButton from '../../atoms/RedirectButton';
+
+function ViewArmyPage({ match }) {
+  const viewArmyID = match.params.ID;
+  const displayArmy = store.getState().userArmies.filter((army) => army.id === viewArmyID)[0];
+
+  return (!displayArmy) ? (<PageLoading />) : (
+    <div data-test="ViewArmyPage" className="container-fluid p-padding text-center">
+      <PageTitle Title="View Army Page" />
+      <RedirectButton redirect={`/armies/edit/${displayArmy.id}`} buttontext="Edit Army" />
+      <TextRow left="Army Name:" right={displayArmy.Name} />
+      <TextRow left="Points:" right={`${displayArmy.Points}`} />
+      <div className="row justify-content-center mt-5">
+        <div className="col-xl-8">
+          <div className="row h4 text-warning">
+            <div className="col-2">Squad</div>
+            <div className="col-2">Unit Qty</div>
+            <div className="col-2">Unit</div>
+            <div className="col-3">Unit Abilities</div>
+            <div className="col-3">Gear</div>
+          </div>
+          {displayArmy.SquadArray.map((roleArray) => (
```

Comment:
The code inside the map block should probably be a component

Reply:
All the of the display pages (army, squads, armies, units, equipment) have custom displays to make the data look as best as I could.  The reason I didn't make it a component was because it's only used once.  What advantage would there be from making it a component?

Reply:
Well smaller files make things more readable and easy to follow. Pretty much whenever I start looping over something complex I always try to move it to smaller nested components. Just easier to maintain and read long term. This file is fine for the size, but larger files (300) shoudl absolutely start breaking things off into smaller components. Although when you hit a 300 line 300 it's something too late and things have to be rewritten to fix the problem.